File: Solutionsin Haskell/p055.hs
{- 
 - Solution to Project Euler problem 55
 - Copyright (c) Sankha Saha (CODR). All rights reserved.
 - 
 - https://github.com/codr07/Project-Euler-Solusions
 -}

import qualified EulerLib


main = putStrLn (show ans)
ans = EulerLib.count (isLychrel 49) [1..9999]

isLychrel 0 n = True
isLychrel iter n = (next /= reverseInt next) && isLychrel (iter - 1) next where
	next = n + (reverseInt n)

reverseInt = revInt 0 where
	revInt acc 0 = acc
	revInt acc n = revInt (acc * 10 + (mod n 10)) (div n 10)

File: codes/python/p055.py
# 
# Solution to Project Euler problem 55
# Copyright (c) Sankha Saha. All rights reserved.
# 
#  
#   
# 


def compute():
	ans = sum(1 for i in range(10000) if is_lychrel(i))
	return str(ans)


def is_lychrel(n):
	for i in range(50):
		n += int(str(n)[ : : -1])
		if str(n) == str(n)[ : : -1]:
			return False
	return True


if __name__ == "__main__":
	print(compute())
